integrationtests/src/test/java/io/kroxylicious/proxy/encyption/EnvelopeEncryptionFilterIT.java

```diff
@@ -0,0 +1,306 @@
+                    n -> n.containsKey(topic));
+
+            // messages produced via Kroxylicious will be encrypted
+            var message = "hello encrypted world";
+            producer.send(new ProducerRecord<>(topic, message)).get(5, TimeUnit.SECONDS);
+
+            // messages produced direct will be plain
+            var plainMessage = "hello plain world";
+            directProducer.send(new ProducerRecord<>(topic, plainMessage)).get(5, TimeUnit.SECONDS);
+
+            consumer.subscribe(List.of(topic));
+            var records = consumer.poll(Duration.ofSeconds(2));
+            assertThat(records.iterator()).toIterable()
+                    .hasSize(2)
+                    .map(ConsumerRecord::value)
+                    .containsExactly(message, plainMessage);
+        }
+    }
+
+    @Test
+    @Disabled("InBandKeyManger doesn't handle nulls")
+    void nullValueRecordProducedAndConsumedSuccessfull(KafkaCluster cluster) throws Exception {
+        var builder = proxy(cluster);
+
+        builder.addToFilters(configureFilter(UUID.randomUUID(), aesGenerator.generateKey()));
+
+        try (var tester = kroxyliciousTester(builder);
+                var admin = tester.admin();
+                var producer = tester.producer();
+                var consumer = tester.consumer()) {
+
+            String topic = tester.createTopic(DEFAULT_VIRTUAL_CLUSTER);
+
+            await().atMost(Duration.ofSeconds(5)).until(() -> admin.listTopics().namesToListings().get(),
+                    n -> n.containsKey(topic));
+
+            String message = null;
+            producer.send(new ProducerRecord<>(topic, message)).get(5, TimeUnit.SECONDS);
+
+            consumer.subscribe(List.of(topic));
+            var records = consumer.poll(Duration.ofSeconds(2));
+            assertThat(records.iterator())
+                    .toIterable()
+                    .singleElement()
+                    .extracting(ConsumerRecord::value)
+                    .isNull();
+        }
+    }
+
+    @Test
+    void produceAndConsumeEncryptedAndPlainTopicsAtSameTime(KafkaCluster cluster, Admin admin) throws Exception {
+        var encryptedTopic = UUID.randomUUID().toString();
+        var plainTopic = UUID.randomUUID().toString();
+
+        var topics = Stream.of(encryptedTopic, plainTopic).map(t -> new NewTopic(t, Optional.empty(), Optional.empty())).toList();
+        admin.createTopics(topics).all().get(5, TimeUnit.SECONDS);
+        await().atMost(Duration.ofSeconds(5)).until(() -> admin.listTopics().namesToListings().get(),
+                n -> n.containsKey(encryptedTopic) && n.containsKey(plainTopic));
+
+        var builder = proxy(cluster);
+
+        UUID keyId = UUID.randomUUID();
+        SecretKey v1 = aesGenerator.generateKey();
+        builder.addToFilters(configureFilter(Map.of(encryptedTopic, keyId), Map.of(keyId,
+                buildMapForKey(v1))));
```

Comment:
I think it would make the test much more readable if we were passing in the YAML for the filter, rather than having to write a load of functions in the test to build the model in code. What makes that tricky is the use of random UUIDs and keys. But this is test code, so I think it would be fine to use a hard coded UUID and key. Wdyt?